scripts/create-darwin-volume.sh

```diff
@@ -112,9 +120,8 @@ main() {
         echo "Creating a Nix Store volume..." >&2
 
         if test_filevault "$disk"; then
-            echo "error: FileVault detected, refusing to create unencrypted volume" >&2
-            echo "See https://nixos.org/nix/manual/#sect-apfs-volume-installation" >&2
-            exit 1
```

Comment:
We still want to fail, however we can direct users to create only the volume and then rerun the installer to perform everything else for them. Given this create volume is a bit of a misnomer, should probably rename the  flag to eg. `--darwin-use-nix-store-volume`.

Reply:
I might be misreading what you mean, but I'm proposing a substantive change on this one; I probably should have called it out with a comment better.

I think @grahamc put a good asterisk on this, though, by pointing out that it would be malpractice not to fail here if the device is _not_ encrypted at rest.

To be explicit, but acknowledge that asterisk, I'm suggesting that in the case of a T2 device with at-rest encryption and FileVault enabled, it may be "better" to go ahead and create the volume for them, but warn them that it isn't encrypted.

- Users who know they want the volume encrypted can encrypt it later (or make sure they pre-encrypt).
- Users who don't know what they want aren't stopped from using the installer and sent to a list of options they may not understand (and may not feel comfortable or be able to make good decisions about without extra support) just because they toggled FileVault during macOS install or from the preferences.

Reply:
FYI the T2 case is already covered. But like mentioned it's a good idea to show a message regarding encryption at rest rather than just silently treating it as non filevault.

Reply:
Just-pushed commit updates how this works to try and address the T2/FileVault condition better and commented a little. I left your function in place for now for reference, but the actual process isn't using it. 

Does this look right-ish?